```
 xrdp/xrdp_process.c:239:46: error: Syntax Error: AST broken, 'self' doesn't have a parent. [internalAstError]
    self->session = libxrdp_init((tbus)self, self->server_trans,
                                             ^
```

This is caused by a typedef which is only accepted by Visual Studio. I filed https://trac.cppcheck.net/ticket/14245 about it.

--- a/common/os_calls.c
+++ b/common/os_calls.c
@@ -96,7 +96,7 @@ struct sockaddr_hvs
 #endif
 
 #include "os_calls.h"
-#include "limits.h"
+#include <limits.h>
 #include "string_calls.h"
 #include "log.h"
 #include "xrdp_constants.h"

--- a/libxrdp/xrdp_surface.c
+++ b/libxrdp/xrdp_surface.c
@@ -22,7 +22,7 @@
 #endif
 
 #include "libxrdp.h"
-#include "freerdp/codec/rfx.h"
+#include <freerdp/codec/rfx.h>
 
 /*****************************************************************************/
 struct xrdp_surface *

--- a/scripts/run_cppcheck.sh
+++ b/scripts/run_cppcheck.sh
@@ -84,7 +84,8 @@ fi
 # Supply default flags passed to cppcheck if necessary
 if [ -z "$CPPCHECK_FLAGS" ]; then
     CPPCHECK_FLAGS="--quiet --force --std=c11 --std=c++11 --inline-suppr \
-                    --enable=warning --error-exitcode=1 -i third_party \
+                    --enable=warning,missingInclude --error-exitcode=1 -i third_party \
+                    --suppress=missingIncludeSystem \
                     --suppress=uninitMemberVar:ulalaca/ulalaca.cpp \
                     --suppress=dangerousTypeCast:ulalaca/XrdpStream.template.cpp \
                     --suppress=shiftTooManyBits:libxrdp/xrdp_mppc_enc.c"

--- a/scripts/run_cppcheck.sh
+++ b/scripts/run_cppcheck.sh
@@ -104,7 +104,23 @@ if [ -z "$CPPCHECK_FLAGS" ]; then
         check_level="Default (option not supported)"
     fi
 
-    CPPCHECK_FLAGS="$CPPCHECK_FLAGS -I . -I common"
+    CPPCHECK_FLAGS="$CPPCHECK_FLAGS \
+                    -I . \
+                    -I common \
+                    -I libipm \
+                    -I libpainter/include \
+                    -I librfxcodec/include \
+                    -I librfxcodec/src \
+                    -I librfxcodec/src/neon \
+                    -I librfxcodec/src/sse2 \
+                    -I libxrdp \
+                    -I sesman/libsesman \
+                    -I sesman/sesexec \
+                    -I third_party \
+                    -I third_party/tomlc99 \
+                    -I xrdp \
+                    -I xrdpapi \
+                    -I xrdpvr"
 fi
 CPPCHECK_FLAGS="$CPPCHECK_FLAGS -D__cppcheck__"
 

--- a/tests/common/test_timers.c
+++ b/tests/common/test_timers.c
@@ -3,7 +3,7 @@
 #include "config_ac.h"
 #endif
 
-#include "limits.h"
+#include <limits.h>
 #include "os_calls.h"
 #include "timers.h"
 

